corehq/apps/geospatial/forms.py

```diff
@@ -57,3 +61,15 @@ def __init__(self, *args, **kwargs):
                 )
             )
         )
+
+    def _get_gps_prop_choices(self):
+        gps_props = get_gps_properties_all_case_types(self.instance.domain)
+
+        # The selected geo case property may be deprecated, so we
+        # should fetch it to ensure it is in the final list
+        gps_props.add(get_geo_case_property(self.instance.domain))
```

Comment:
I'm not entirely sure how we could squash this down to a single query. We need to get the list of non-deprecated GPS properties, but we also need to find out what the user has set as the custom case property in their settings (this may not be the default `GEO_POINT_CASE_PROPERTY`). This is so we can ensure it is in the final list, even if the selected property is deprecated. Without doing the second query, I'm uncertain how we would achieve that. 

@mkangia do you have any suggestions that I might not be thinking of?

Reply:
Hey @zandre-eng 
I noted down the steps later in this comment, https://github.com/dimagi/commcare-hq/pull/33426#discussion_r1315056505

I am saying to fetch all properties for the domain and then filter in django itself.

Reply:
Related to [this](https://github.com/dimagi/commcare-hq/pull/33426#discussion_r1317311815) comment, the warning logic has been moved to JS and now only a single query is being run.